File: java/src/test/java/it/giannibombelli/wsc2026/testsupport/events/CapturingEventPublisher.java
package it.giannibombelli.wsc2026.testsupport.events;

import it.giannibombelli.wsc2026.common.application.events.EventPublisher;

import java.util.ArrayList;
import java.util.List;

public final class CapturingEventPublisher<T> implements EventPublisher<T> {
    private final List<T> events = new ArrayList<>();

    @Override
    public void publish(T event) {
        if (event == null) {
            throw new IllegalArgumentException("event must not be null");
        }
        events.add(event);
    }

    public List<T> events() {
        return List.copyOf(events);
    }
}

File: typescript/tests/testsupport/events/capturingEventPublisher.ts
// Publisher di test che cattura gli eventi pubblicati.

import type { EventPublisher } from '@/common/application/events/eventPublisher.js';

export class CapturingEventPublisher<T> implements EventPublisher<T> {
  private readonly _events: T[] = [];

  publish(event: T): void {
    if (event === null || event === undefined) throw new Error('event must be defined');
    this._events.push(event);
  }

  events(): T[] {
    return [...this._events];
  }
}
